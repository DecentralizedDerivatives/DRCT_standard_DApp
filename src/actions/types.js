export const SET_USER_ACCOUNT = 'SET_USER_ACCOUNT';
export const SET_USER_BALANCE = 'SET_USER_BALANCE';
export const SET_USER_TRANSACTIONS = 'SET_USER_TRANSACTIONS';
export const SET_USER_POSITIONS = 'SET_USER_POSITIONS';
export const SET_CONTRACT_DETAILS = 'SET_CONTRACT_DETAILS';
export const SET_ORDER_DETAILS = 'SET_ORDER_DETAILS';
export const SET_SELECTED_TOKEN = 'SET_SELECTED_TOKEN';

export const SET_IS_CONNECTED = 'SET_IS_CONNECTED';
export const TX_PROCESSING = 'TEX_PROCESSING';
export const SET_TX_RECEIPT = 'SET_TX_RECEIPT';
export const RESET_TX_STATE = 'RESET_TX_STATE';
export const SET_PROCESSING_ERROR = 'SET_PROCESSING_ERROR';
export const CLEAR_PROCESSING_ERROR = 'CLEAR_PROCESSING_ERROR';
