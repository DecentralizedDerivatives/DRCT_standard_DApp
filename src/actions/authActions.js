import { web3 } from '../ethereum';

import { GET_ERRORS, SET_USER_CONNECTED } from './types';

// Check connection status 
export const checkConnection = () => dispatch => {
  try {
    const accounts = await web3.eth.getAccounts();
    const network = await web3.eth.net.getId();
    
    dispatch({
      type: SET_USER_CONNECTED,
      payload: (accounts.length && network === 4)  
      })
    }
  } catch(err) {
    dispatch({
      type: GET_ERRORS,
      payload: err.response.data
    })
  }
}
