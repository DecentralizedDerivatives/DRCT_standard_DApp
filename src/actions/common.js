import { Oracle } from '../ethereum';

const moment = require('moment');

export const getStartDatePrice = async (oracleAddress, startDate) => {
  let dt = startDate ? moment(startDate, 'MM/DD/YYYY').utc().startOf('day') : null;
  let date = dt.format('x') / 1000
  const oracle = await Oracle.at(oracleAddress)
  var data = await oracle.retrieveData(date)
  if (!data || !data.c || !data.c[0]) { return 0 }
  let val = parseFloat('0.00' + data.c[0].toString())
  // console.log('val', val)
  let multiplier = '1e' + (data.e).toString()
  let adjustedValue = val * (Number(multiplier))
  // console.log('adjustedValue', adjustedValue)
  return adjustedValue
}
