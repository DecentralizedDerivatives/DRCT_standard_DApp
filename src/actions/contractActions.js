import { Factory, Exchange, DRCT, Oracle } from '../ethereum';
import {
  SET_CONTRACT_DETAILS,
  SET_CONTRACT_OPEN_DATES,
  SET_CONTRACT_START_PRICE,
  SET_ORDERBOOK,
  SET_FETCH_IN_PROGRESS,
  REMOVE_FETCH_IN_PROGRESS,
  SET_FETCHING_ERROR,
  SET_RECENT_TRADES
} from './types';

import FactoryProvider from '../factoryProvider';
const moment = require('moment');

export const getContractDetails = (symbol) => async dispatch => {
  try {
    const provider = FactoryProvider.getFromSymbol(symbol);
    // const factory = await Factory.at(provider && provider.address ? provider.address : '');
    // const response = await factory.getVariables();
    const details = {
      contractAddress: provider.address,
      contractDuration: provider.duration,
      contractMultiplier: provider.multiplier,
      oracleAddress: provider.oracle
      // contractAddress: response[0],
      // contractDuration: response[1].c[0],
      // contractMultiplier: response[2].c[0],
      // oracleAddress: response[3]
    };

    dispatch({
      type: SET_CONTRACT_DETAILS,
      payload: details
    });
    return details;
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: err.message.split('\n')[0]
    });
  }
};

export const getStartDatePrice = (oracleAddress, startDate) => async dispatch => {
  try {
    const oracle = await Oracle.at(oracleAddress)
    var data = await oracle.retrieveData(startDate);
    dispatch({
      type: SET_CONTRACT_START_PRICE,
      payload: data.c[0]
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: err.message.split('\n')[0]
    });
  }
}
export const getOrderBook = (isSilent) => async dispatch => {
  try {
    if (!isSilent) { dispatch({ type: SET_FETCH_IN_PROGRESS, payload: SET_ORDERBOOK }); };
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);
    let numBooks = await exchange.getBookCount();
    var factories = FactoryProvider.factories();
    let _allrows = [];
    for (let i = 0; i < numBooks; i++) {
      let book = await exchange.openBooks(i);
      // console.log('book', book)
      for (var p = 0; p < factories.length; p++) {
        const factory = await Factory.at(factories[p].address);
        // console.log('factory', factories[p].symbol);
        let tokenDate = await factory.token_dates.call(book);
        // console.log('tokenDate', tokenDate)
        if (tokenDate.c[0] === 0) { continue }
        let orders = await exchange.getOrders(book);
        // console.log('orders', orders)
        for (let j = 0; j < orders.length; j++) {
          if (orders[j].c[0] > 0) {
            let order = await exchange.getOrder(orders[j].c[0]);
            // console.log('order', order)
            let tokenType = (await factory.getTokenType(order[3])).c[0];
            let date = new Date(tokenDate.c[0] * 1000);
            let startDate = new Date();
            startDate.setDate(date.getDate() + 1);
            var todayMinusSixDays = new Date();
            todayMinusSixDays.setDate(todayMinusSixDays.getDate() - 6);
            if (date > todayMinusSixDays) {
              let orderDate = startDate.getUTCMonth() + 1 + '/' +
                startDate.getUTCDate() + '/' + startDate.getUTCFullYear();
              var precisePrice = parseFloat(order[1].c[0]/10000).toFixed(5);
              _allrows.push({
                orderId: orders[j].c[0].toString(),
                address: order[3],
                price: precisePrice,
                quantity: order[2].c[0].toString(),
                date: orderDate.toString(),
                symbol: factories[p].symbol,
                tokenType: tokenType === 1 ? 'Short' : 'Long'
               });
            }
          }
        }
      }
      _allrows.sort(function (a, b) {
        return a.orderId - b.orderId;
      });
    }
    dispatch({
      type: SET_ORDERBOOK,
      payload: _allrows
    });
    dispatch({ type: REMOVE_FETCH_IN_PROGRESS, payload: SET_ORDERBOOK });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'Order Book: ' + err.message.split('\n')[0]
    });
  }
};

export const getRecentTrades = (isSilent) => async dispatch => {
  try {
    if (!isSilent) { dispatch({ type: SET_FETCH_IN_PROGRESS, payload: SET_RECENT_TRADES }); };
    var staticAddresses = FactoryProvider.getStaticAddresses();
    const exchange = await Exchange.at(staticAddresses.exchange);

    let transferEvent = await exchange.Sale(
      {},
      { fromBlock: 0, toBlock: 'latest' }
    );

    transferEvent.get(async function (err, events) {
      // console.log('events', events);
      var trades = [];
      if (events.length > 0) {
        for (let i = events.length - 1; i >= Math.max(events.length - 10, 0); i--) {
          var token = events[i].args['_token'].toString();
          // console.log('token', token);
          var drct = DRCT.at(token);
          // console.log('drct', drct);
          var factoryAddress = await drct.getFactoryAddress();
          const factory = await Factory.at(factoryAddress);
          let tokenType = (await factory.getTokenType(token)).c[0];
          // console.log('factoryAddress', factoryAddress);
          var provider = FactoryProvider.getFromAddress(factoryAddress);
          var precisePrice = parseFloat(events[i].args['_price']/1e18).toFixed(5);
          trades.push({
            address: token,
            volume: events[i].args['_amount'].toString(),
            price: precisePrice,
            contractDuration: provider && provider.duration ? provider.duration : 0,
            contractMultiplier: provider && provider.multiplier ? provider.multiplier : 0,
            symbol: provider && provider.symbol ? provider.symbol : '??',
            tokenType: tokenType === 1 ? 'Short' : 'Long'
          });
        }
      }
      dispatch({
        type: SET_RECENT_TRADES,
        payload: trades
      });
      dispatch({ type: REMOVE_FETCH_IN_PROGRESS, payload: SET_RECENT_TRADES });
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'Recent Trades: ' + err.message.split('\n')[0]
    });
  }
};

export const getContractOpenDates = (currencyAddress) => async dispatch => {
  try {
    const factory = await Factory.at(currencyAddress);
    // console.log(factory);
    let openDates = {};
    const numDates = await factory.getDateCount();

    for (let i = 0; i < numDates; i++) {
      const startDate = (await factory.startDates.call(i)).c[0];
      let formattedDate = new Date(startDate * 1000);
      var momentDate = moment(formattedDate);
      var pastDate = moment().subtract(5, 'days').startOf('day');
      if (momentDate.isBefore(pastDate)) { continue }
      formattedDate = formattedDate.getUTCMonth() + 1 + '/' +
        formattedDate.getUTCDate() + '/' +
        formattedDate.getUTCFullYear();
      openDates[startDate] = formattedDate;
    }
    dispatch({
      type: SET_CONTRACT_OPEN_DATES,
      payload: openDates
    });
  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'Contract Open Dates: ' + err.message.split('\n')[0]
    });
  }
};
