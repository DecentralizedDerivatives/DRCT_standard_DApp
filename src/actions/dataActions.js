import api from '../api';
import { SET_PRICECHART_DATA, SET_FETCHING_ERROR } from './types';

export const getPriceChartData = type => async dispatch => {
  try {
    const data = await api[type].get();
      
    dispatch({
      type: SET_PRICECHART_DATA,
      payload: data 
    });      

  } catch (err) {
    dispatch({
      type: SET_FETCHING_ERROR,
      payload: 'PriceChart: ' + err.message.split('\n')[0]
    });
  }
};
