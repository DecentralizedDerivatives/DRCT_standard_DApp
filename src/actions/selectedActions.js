import { SET_SELECTED_TOKEN, CLEAR_SELECTED_TOKEN } from './types';

export const setSelectedToken = token => dispatch => {
  dispatch({
    type: SET_SELECTED_TOKEN,
    payload: token
  });
};

export const clearSelectedToken = token => dispatch => {
  dispatch({
    type: CLEAR_SELECTED_TOKEN,
    payload: ''
  });
};
