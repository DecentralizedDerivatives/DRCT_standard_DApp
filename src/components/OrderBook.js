import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Table } from 'reactstrap';
import { getOrderBook } from '../actions/contractActions';

// Use named export for unconnected component for testing
export class OrderBook extends Component {
  renderRows = () => {
    if (this.props.orderbook.length === 0) {
      return <tr><td colSpan='12' style={{textAlign: 'center'}}><h5>No Recent Orders</h5></td></tr>
    }
    var rows = this.props.orderbook.map(order => {
      const { orderId, address, price, quantity, date, symbol } = order;
      return (
        <tr key={orderId} className='clickable' onClick={this.props.onRowClick.bind(this, address, symbol)}>
          <td>{orderId}</td>
          <td>{symbol}</td>
          <td>{price}</td>
          <td>{quantity}</td>
          <td>{date}</td>
        </tr>
      );
    });
    return rows;
  };

  render() {
    return (
      <div className="container">
        <div className="order-book">
          <Table
            id="positions-table"
            className="table table-hover table-striped table-responsive"
          >
            <thead>
              <tr>
                <th>Order Book</th>
              </tr>
              <tr>
                <th>Order Id</th>
                <th>Asset</th>
                <th>Price (ETH)</th>
                <th>Quantity</th>
                <th>Start Date</th>
              </tr>
            </thead>
            <tbody>{this.renderRows()}</tbody>
          </Table>
        </div>
      </div>
    );
  }
}

OrderBook.propTypes = {
  onRowClick: PropTypes.func.isRequired,
  orderbook: PropTypes.array.isRequired,
  contractDuration: PropTypes.number.isRequired,
  contractMultiplier: PropTypes.number.isRequired
};

const mapStateToProps = state => ({
  orderbook: state.contract.orderbook,
  contractDuration: state.contract.contractDuration,
  contractMultiplier: state.contract.contractMultiplier
});

export default connect(
  mapStateToProps,
  { getOrderBook }
)(OrderBook);
