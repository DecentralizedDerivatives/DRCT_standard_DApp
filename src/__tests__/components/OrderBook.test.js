import OrderBook from '../../components/OrderBook';
import { SET_FETCH_IN_PROGRESS, SET_ORDERBOOK } from '../../actions/types';

// TODO: setup
describe('<OrderBook />', () => {
  it('renders the component', () => {
    const onRowClick = jest.fn();
    const refreshPage = jest.fn();

    const wrapper = shallow(
      <OrderBook
        store={initFixtureStore()}
        onRowClick={onRowClick}
        refreshPage={refreshPage}
      />
    ).dive();

    wrapper
      .find('button[children="Buy"]')
      .first()
      .simulate('click', {
        preventDefault: jest.fn(),
        stopPropagation: jest.fn(),
      });

    wrapper
      .find('button[children="Unlist"]')
      .first()
      .simulate('click', {
        preventDefault: jest.fn(),
        stopPropagation: jest.fn(),
      });

    wrapper.setProps({ buyOrderTx: undefined });
    wrapper.setProps({ buyOrderTx: '0x000...' });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty component', () => {
    const onRowClick = jest.fn();

    const wrapper = shallow(
      <OrderBook store={initStore()} onRowClick={onRowClick} />
    ).dive();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders loading component', () => {
    const onRowClick = jest.fn();
    const store = initFixtureStore();

    store.dispatch({ type: SET_FETCH_IN_PROGRESS, payload: SET_ORDERBOOK });

    const wrapper = shallow(
      <OrderBook store={store} onRowClick={onRowClick} />
    ).dive();

    expect(wrapper).toMatchSnapshot();
  });

  it('formats money', () => {
    const onRowClick = jest.fn();

    const wrapper = shallow(
      <OrderBook store={initStore()} onRowClick={onRowClick} />
    ).dive();

    expect(wrapper.instance().formatMoney()).toMatchSnapshot();
    expect(wrapper.instance().formatMoney(undefined, '--')).toMatchSnapshot();
    expect(wrapper.instance().formatMoney(-1)).toMatchSnapshot();
    expect(wrapper.instance().formatMoney(1)).toMatchSnapshot();
  });

  it('formats percentage', () => {
    const onRowClick = jest.fn();

    const wrapper = shallow(
      <OrderBook store={initStore()} onRowClick={onRowClick} />
    ).dive();

    expect(wrapper.instance().formatPercent()).toMatchSnapshot();
    expect(wrapper.instance().formatPercent(undefined, '--')).toMatchSnapshot();
    expect(wrapper.instance().formatPercent(-1)).toMatchSnapshot();
    expect(wrapper.instance().formatPercent(1)).toMatchSnapshot();
  });
});
