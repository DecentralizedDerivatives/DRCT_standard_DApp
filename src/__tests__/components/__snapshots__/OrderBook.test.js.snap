// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderBook /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderBook
    buyOrderError={null}
    contractDuration={0}
    contractMultiplier={0}
    getOrderBook={[Function]}
    loading={false}
    orderbook={Array []}
    sendBuyOrder={[Function]}
    sendUnlistOrder={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="table table-hover table-striped table-responsive"
      >
        <thead>
          <tr>
            <th
              colSpan="6"
            >
              Order Book
              <div
                className="warning"
              />
            </th>
          </tr>
          <tr>
            <th
              style={
                Object {
                  "width": "15%",
                }
              }
            >
              Order Id
            </th>
            <th
              style={
                Object {
                  "width": "40%",
                }
              }
            >
              Asset
            </th>
            <th
              style={
                Object {
                  "width": "15%",
                }
              }
            >
              Price (ETH)
            </th>
            <th
              style={
                Object {
                  "width": "15%",
                }
              }
            >
              Quantity
            </th>
            <th>
              Start Date
            </th>
            <th>
              Gain/Loss
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              colSpan="12"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <h5>
                No Recent Orders
              </h5>
            </td>
          </tr>
        </tbody>
      </table>,
      "className": "table-container order-book",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th
                colSpan="6"
              >
                Order Book
                <div
                  className="warning"
                />
              </th>
            </tr>
            <tr>
              <th
                style={
                  Object {
                    "width": "15%",
                  }
                }
              >
                Order Id
              </th>
              <th
                style={
                  Object {
                    "width": "40%",
                  }
                }
              >
                Asset
              </th>
              <th
                style={
                  Object {
                    "width": "15%",
                  }
                }
              >
                Price (ETH)
              </th>
              <th
                style={
                  Object {
                    "width": "15%",
                  }
                }
              >
                Quantity
              </th>
              <th>
                Start Date
              </th>
              <th>
                Gain/Loss
              </th>
              <th />
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td
                colSpan="12"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <h5>
                  No Recent Orders
                </h5>
              </td>
            </tr>
          </tbody>,
        ],
        "className": "table table-hover table-striped table-responsive",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <th
                  colSpan="6"
                >
                  Order Book
                  <div
                    className="warning"
                  />
                </th>
              </tr>,
              <tr>
                <th
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  Order Id
                </th>
                <th
                  style={
                    Object {
                      "width": "40%",
                    }
                  }
                >
                  Asset
                </th>
                <th
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  Price (ETH)
                </th>
                <th
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  Quantity
                </th>
                <th>
                  Start Date
                </th>
                <th>
                  Gain/Loss
                </th>
                <th />
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <th
                  colSpan="6"
                >
                  Order Book
                  <div
                    className="warning"
                  />
                </th>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Order Book",
                    <div
                      className="warning"
                    />,
                  ],
                  "colSpan": "6",
                },
                "ref": null,
                "rendered": Array [
                  "Order Book",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                      "className": "warning",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "th",
              },
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    style={
                      Object {
                        "width": "15%",
                      }
                    }
                  >
                    Order Id
                  </th>,
                  <th
                    style={
                      Object {
                        "width": "40%",
                      }
                    }
                  >
                    Asset
                  </th>,
                  <th
                    style={
                      Object {
                        "width": "15%",
                      }
                    }
                  >
                    Price (ETH)
                  </th>,
                  <th
                    style={
                      Object {
                        "width": "15%",
                      }
                    }
                  >
                    Quantity
                  </th>,
                  <th>
                    Start Date
                  </th>,
                  <th>
                    Gain/Loss
                  </th>,
                  <th />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Order Id",
                    "style": Object {
                      "width": "15%",
                    },
                  },
                  "ref": null,
                  "rendered": "Order Id",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Asset",
                    "style": Object {
                      "width": "40%",
                    },
                  },
                  "ref": null,
                  "rendered": "Asset",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Price (ETH)",
                    "style": Object {
                      "width": "15%",
                    },
                  },
                  "ref": null,
                  "rendered": "Price (ETH)",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Quantity",
                    "style": Object {
                      "width": "15%",
                    },
                  },
                  "ref": null,
                  "rendered": "Quantity",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Start Date",
                  },
                  "ref": null,
                  "rendered": "Start Date",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Gain/Loss",
                  },
                  "ref": null,
                  "rendered": "Gain/Loss",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <td
                colSpan="12"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <h5>
                  No Recent Orders
                </h5>
              </td>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <td
                colSpan="12"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <h5>
                  No Recent Orders
                </h5>
              </td>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h5>
                  No Recent Orders
                </h5>,
                "colSpan": "12",
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No Recent Orders",
                },
                "ref": null,
                "rendered": "No Recent Orders",
                "type": "h5",
              },
              "type": "td",
            },
            "type": "tr",
          },
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="table table-hover table-striped table-responsive"
        >
          <thead>
            <tr>
              <th
                colSpan="6"
              >
                Order Book
                <div
                  className="warning"
                />
              </th>
            </tr>
            <tr>
              <th
                style={
                  Object {
                    "width": "15%",
                  }
                }
              >
                Order Id
              </th>
              <th
                style={
                  Object {
                    "width": "40%",
                  }
                }
              >
                Asset
              </th>
              <th
                style={
                  Object {
                    "width": "15%",
                  }
                }
              >
                Price (ETH)
              </th>
              <th
                style={
                  Object {
                    "width": "15%",
                  }
                }
              >
                Quantity
              </th>
              <th>
                Start Date
              </th>
              <th>
                Gain/Loss
              </th>
              <th />
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                colSpan="12"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <h5>
                  No Recent Orders
                </h5>
              </td>
            </tr>
          </tbody>
        </table>,
        "className": "table-container order-book",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th
                  colSpan="6"
                >
                  Order Book
                  <div
                    className="warning"
                  />
                </th>
              </tr>
              <tr>
                <th
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  Order Id
                </th>
                <th
                  style={
                    Object {
                      "width": "40%",
                    }
                  }
                >
                  Asset
                </th>
                <th
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  Price (ETH)
                </th>
                <th
                  style={
                    Object {
                      "width": "15%",
                    }
                  }
                >
                  Quantity
                </th>
                <th>
                  Start Date
                </th>
                <th>
                  Gain/Loss
                </th>
                <th />
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td
                  colSpan="12"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <h5>
                    No Recent Orders
                  </h5>
                </td>
              </tr>
            </tbody>,
          ],
          "className": "table table-hover table-striped table-responsive",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <th
                    colSpan="6"
                  >
                    Order Book
                    <div
                      className="warning"
                    />
                  </th>
                </tr>,
                <tr>
                  <th
                    style={
                      Object {
                        "width": "15%",
                      }
                    }
                  >
                    Order Id
                  </th>
                  <th
                    style={
                      Object {
                        "width": "40%",
                      }
                    }
                  >
                    Asset
                  </th>
                  <th
                    style={
                      Object {
                        "width": "15%",
                      }
                    }
                  >
                    Price (ETH)
                  </th>
                  <th
                    style={
                      Object {
                        "width": "15%",
                      }
                    }
                  >
                    Quantity
                  </th>
                  <th>
                    Start Date
                  </th>
                  <th>
                    Gain/Loss
                  </th>
                  <th />
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <th
                    colSpan="6"
                  >
                    Order Book
                    <div
                      className="warning"
                    />
                  </th>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Order Book",
                      <div
                        className="warning"
                      />,
                    ],
                    "colSpan": "6",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Order Book",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": null,
                        "className": "warning",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                  ],
                  "type": "th",
                },
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th
                      style={
                        Object {
                          "width": "15%",
                        }
                      }
                    >
                      Order Id
                    </th>,
                    <th
                      style={
                        Object {
                          "width": "40%",
                        }
                      }
                    >
                      Asset
                    </th>,
                    <th
                      style={
                        Object {
                          "width": "15%",
                        }
                      }
                    >
                      Price (ETH)
                    </th>,
                    <th
                      style={
                        Object {
                          "width": "15%",
                        }
                      }
                    >
                      Quantity
                    </th>,
                    <th>
                      Start Date
                    </th>,
                    <th>
                      Gain/Loss
                    </th>,
                    <th />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Order Id",
                      "style": Object {
                        "width": "15%",
                      },
                    },
                    "ref": null,
                    "rendered": "Order Id",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Asset",
                      "style": Object {
                        "width": "40%",
                      },
                    },
                    "ref": null,
                    "rendered": "Asset",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Price (ETH)",
                      "style": Object {
                        "width": "15%",
                      },
                    },
                    "ref": null,
                    "rendered": "Price (ETH)",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Quantity",
                      "style": Object {
                        "width": "15%",
                      },
                    },
                    "ref": null,
                    "rendered": "Quantity",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Start Date",
                    },
                    "ref": null,
                    "rendered": "Start Date",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Gain/Loss",
                    },
                    "ref": null,
                    "rendered": "Gain/Loss",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <td
                  colSpan="12"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <h5>
                    No Recent Orders
                  </h5>
                </td>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <td
                  colSpan="12"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <h5>
                    No Recent Orders
                  </h5>
                </td>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h5>
                    No Recent Orders
                  </h5>,
                  "colSpan": "12",
                  "style": Object {
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No Recent Orders",
                  },
                  "ref": null,
                  "rendered": "No Recent Orders",
                  "type": "h5",
                },
                "type": "td",
              },
              "type": "tr",
            },
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
